woocommerce/woocommerce-android
# Issue/2410 add product detail bottom sheet

Fixes #2410 by adding a bottom sheet in product detail screen to add new fields to the product such as categories, tags and short description.

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/87293400-e2537600-c51f-11ea-8763-4233dedfc72f.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/87293404-e41d3980-c51f-11ea-879e-e02a209066b6.png" width="300"/>

As mentioned in #2410, the bottom sheet would be displayed for the following product fields, based on the product type: 
**Physical product**
- Shipping
- Product categories
- Product tags
- Short description

**Grouped product**
- Product categories
- Product tags
- Short description

**Affiliate product**
- Product categories
- Product tags
- Short description

#### Notes
- The bottom sheet only displays the list of fields that are currently editable in the app. It does not include fields such as up sells, cross sells etc.
- This feature is only enabled for the debug build.
- The bottom sheet displays the list of fields mentioned above, only if those fields are empty. For instance, if there are no tags associated with the product, the bottom sheet includes the option to add tags to the product.
- This PR is in draft till this [PR](https://github.com/woocommerce/woocommerce-android/pull/2602) and this [PR](https://github.com/woocommerce/woocommerce-android/pull/2603) can be reviewed and merged.

#### Testing
- Click on a physical product (simple/virtual) that does not have any shipping info or tags or categories. 
  - Click on the "Add more details" button.
  - Ensure that the shipping item is displayed in the list.
  - Click on the shipping item and update the shipping field. 
  - Click on the Done button.
  - Verify that the shipping section is now displayed correctly for the product.
  - Click on "Add more details" button and verify that the shipping item is no longer displayed in the bottom sheet.
- Click on a grouped product that does not have any tags info. 
  - Click on the "Add more details" button.
  - Ensure that the tags item is displayed in the list.
  - Click on the tags item and update the tags field. 
  - Click on the Done button.
  - Verify that the tags section is now displayed correctly for the product.
  - Click on "Add more details" button and verify that the tags item is no longer displayed in the bottom sheet.
- Click on an external product that does not have any short description info. 
  - Click on the "Add more details" button.
  - Ensure that the short description item is displayed in the list.
  - Click on the short description item and update the short description field. 
  - Click on the Done button.
  - Verify that the short description section is now displayed correctly for the product.
  - Click on "Add more details" button and verify that the short description item is no longer displayed in the bottom sheet.

Update release notes:
